test/library/packages/Shlex/Shlex.chpl

```diff
@@ -0,0 +1,442 @@
+                                      if the argument `punctuation_chars` is
+                                      set to true. Defaults to `();|&`.
+      :type custom_punctuation_chars: `string`
+    */
+    proc init(instream:string, posix: bool = false,
+              punctuation_chars: bool = false, commenters: string = '#',
+              custom_punctuation_chars: string = '();<>|&') {
+      this.instream = instream.strip(trailing=true);
+      this.posix = posix;
+      this.lineno = 1;
+      this.debug = 0;
+      this.commenters = commenters;
+      this.wordchars = "abcdfeghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+      this.wordchars += "0123456789_";
+      if posix {
+        this.wordchars += "ßàáâãäåæçèéêëìíîïðñòóôõöøùúûüýþÿÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐ";
+        this.wordchars += "ÑÒÓÔÕÖØÙÚÛÜÝÞ";
+      }
+      this.whitespace = " \t\n\r";
+      this.whitespace_split = false;
+      this.quotes = "\"'";
+      this.escapse = "\\";
+      this.escapsedquotes = '"';
+      this.state = " ";
+      this.token = '';
+      this.tokindex = 0;
+      if (punctuation_chars == false) {
+        this.punctuation_chars = '';
+      }
+      else{
+        this.punctuation_chars = if custom_punctuation_chars == '' then '();<>|&'
+        else custom_punctuation_chars;
+        this.wordchars += '~-./?*=';
+        for c in this.punctuation_chars {
+          this.wordchars = this.wordchars.replace(c, '');
+        }
+      }
+    }
+
+    /*
+      Returns the next token from `instream`. Returns empty string when EOF
+      is reached.
+
+      :returns: Next token from `instream`.
+      :rtype: `string`
+
+    */
+    proc get_token(): string {
+      if !this.pushback.isEmpty() {
+        var tok = this.pushback.pop(0);
+        if this.debug >= 1 {
+          writeln("shlex: popping token " + tok);
+        }
+        return tok;
+      }
+      if(this.tokindex == -1){
+        return '';
+      }
+      var raw = try! this.read_token();
+      if this.debug >= 1 {
+        if(raw != ''){
+          writeln("shlex: token = ", raw);
+        }
+        else{
+          writeln("shlex: token = EOF");
+        }
+      }
+      return raw;
+    }
+
+    /* 
+      Parses through `instream` and returns the next token.
+      
+      :returns: Next token from `instream`.
+      :rtype: `string`
+      :throws Error: Unbalanced quotes or absence of escaped character.
+    */
+    proc read_token(): string throws{
+      var quoted:bool = false;
+      var escapsedstate:string = ' ';
+      var nextchar:string = '';
+      while true {
+        if this.tokindex == -1 then break;
+        if (this.punctuation_chars != '' && !this._pushback_chars.isEmpty()){
+          nextchar = this._pushback_chars.pop();
+        }
+        else if(this.tokindex < this.instream.size){
+          nextchar = this.instream[this.tokindex];
+          this.tokindex += 1;
+        }
+        else{
```

Comment:
```suggestion
        else {
```

Reply:
Note: I _think_ committing these suggestions will generate an error in the DCO signed author checks, so I believe it's better for you to make the changes yourself in the code.